cmd/server/pactasrv/populate.go

```diff
@@ -0,0 +1,104 @@
+package pactasrv
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/RMI/pacta/oapierr"
+	"github.com/RMI/pacta/pacta"
+	"go.uber.org/zap"
+)
+
+func populateAll[Source any, TargetID ~string, Target any](
```

Comment:
This is a clever use of generics, but it's also hard to understand what it's doing, both here and where its called. I'd add a comment or similar.

Also, it looks like we're doing that "populate a pre-existing pointer" thing, which I begrudgingly accepted was useful in a GraphQL system for populating arbitrary recursive responses, but in a standard RESTful server, I can't see the motivation for such gymnastics when populating a static, non-recursive structure.

Reply:
GC on commenting this to make it clearer. 

Not sure of the distinction you're drawing on that one: we have recursive structures in our data model here too: `Portfolio.MemberOf` is a `PortfolioGroupMembership`,which has a `Portfolio` field (same with `PortfolioGroup`). The value of these selsective population things is exactly that we can build a DB layer that is "dumb", and then populate the data we want selectively based on the context.

This is the ~the same pattern we used for Abound, ex, because the data model is actually really similar. Portfolio:Art, Gallery:PortfolioGroup, ex.